### PR TITLE
Document data structures

### PR DESCRIPTION
**Pull request by tmpolaczyk**
Add some missing lints to `data_structures/src/lib.rs`. This includes documenting every public field, which may take a while.

### CODE REVIEW

File: data_structures/src/chain.rs [hunk 1]
```diff
@@ -1129,14 +1177,18 @@ impl DataRequestOutput {
...
+    /// Signature
     pub signature: Signature,
+    /// Public key
     pub public_key: PublicKey,
 }
 
 /// Public Key data structure
 #[derive(Debug, Default, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
 pub struct PublicKey {
+    /// Compressed
```
Comment (girazoki) [inline data_structures/src/chain.rs:1189]:
```suggestion
    /// Byte indicating how to decompress the public key
```

File: data_structures/src/chain.rs [hunk 2]
```diff
@@ -1129,14 +1177,18 @@ impl DataRequestOutput {
...
     pub compressed: u8,
+    /// Bytes
```
Comment (girazoki) [inline data_structures/src/chain.rs:1191]:
```suggestion
    /// Public key bytes
```

File: data_structures/src/vrf.rs [hunk 3]
```diff
@@ -70,10 +70,12 @@ impl VrfProof {
             .verify(&self.public_key.to_bytes(), &self.proof, &message.0)?)
     }
 
+    /// Return the pkh of the identity that signed the proof
     pub fn pkh(&self) -> PublicKeyHash {
         PublicKeyHash::from_public_key(&self.public_key)
     }
 
+    /// Clone the proof
```
Comment (girazoki) [inline data_structures/src/vrf.rs:78]:
```suggestion
    // Getter for the VRF proof
```

File: data_structures/src/chain.rs (hunk 1 above)
Comment (mariocao) [reply data_structures/src/chain.rs:1189]: Shame... :rofl:

**Comment by lrubiorod** [comment]
@tmpolaczyk when this PR?

**Comment by tmpolaczyk** [comment]
It's taking longer than expected, I think we can merge the existing changes and open an issue to document the remaining items.

**Review by aesedepece — APPROVED** [review]

**Pull request merged**